battstat/battstat-upower.c

```diff
@@ -133,6 +131,7 @@ battstat_upower_cleanup( void )
   if( upc == NULL )
     return;
   
+  g_object_unref( devices );
```

Comment:
I'd re-do the code here to put `unref` calls into `if (ptr != NULL) ... unref()`
Besides, call `g_ptr_array_unref` for safety. Hell knows how GTK works.